src/System.Private.CoreLib/shared/System/MemoryExtensions.Fast.cs

```diff
@@ -179,6 +179,50 @@ public static int IndexOf(this ReadOnlySpan<char> span, ReadOnlySpan<char> value
             return -1;
         }
 
+        /// <summary>
+        /// Reports the zero-based index of the last occurrence of the specified <paramref name="value"/> in the current <paramref name="span"/>.
+        /// <param name="span">The source span.</param>
+        /// <param name="value">The value to seek within the source span.</param>
+        /// <param name="comparisonType">One of the enumeration values that determines how the <paramref name="span"/> and <paramref name="value"/> are compared.</param>
+        /// </summary>
+        public static int LastIndexOf(this ReadOnlySpan<char> span, ReadOnlySpan<char> value, StringComparison comparisonType)
+        {
+            string.CheckStringComparison(comparisonType);
+
+            if (value.Length == 0)
+            {
+                return 0;
+            }
+
+            if (span.Length == 0)
+            {
+                return -1;
+            }
+
+            if (GlobalizationMode.Invariant)
+            {
+                return CompareInfo.InvariantIndexOf(span, value, string.GetCaseCompareOfComparisonCulture(comparisonType) != CompareOptions.None, start: false);
+            }
+
+            switch (comparisonType)
+            {
+                case StringComparison.CurrentCulture:
+                case StringComparison.CurrentCultureIgnoreCase:
+                    return CultureInfo.CurrentCulture.CompareInfo.IndexOf(span, value, string.GetCaseCompareOfComparisonCulture(comparisonType), start: false);
+
+                case StringComparison.InvariantCulture:
+                case StringComparison.InvariantCultureIgnoreCase:
+                    return CompareInfo.Invariant.IndexOf(span, value, string.GetCaseCompareOfComparisonCulture(comparisonType), start: false);
+
+                case StringComparison.Ordinal:
+                case StringComparison.OrdinalIgnoreCase:
+                    return CompareInfo.Invariant.IndexOfOrdinal(span, value, string.GetCaseCompareOfComparisonCulture(comparisonType) != CompareOptions.None, start: false);
+            }
+
+            Debug.Fail("StringComparison outside range");
+            return -1;
```

Comment:
We already validated at the beginning that the StringComparison was in range, so this is dead code.  You can instead, for example, change the code:
```C#
case StringComparison.Ordinal:
case StringComparison.OrdinalIgnoreCase:
    return CompareInfo.Invariant.IndexOfOrdinal(...);
```
to instead be something like:
```C#
default:
    Debug.Assert(comparisonType == StringComparison.Ordinal || comparisonType == StringComparison.OrdinalIgnoreCase);
    return CompareInfo.Invariant.IndexOfOrdinal(...);
```